x-pack/solutions/observability/plugins/apm/server/routes/service_map/get_trace_sample_ids.ts

```diff
@@ -74,13 +70,118 @@ export async function getTraceSampleIds({
 
   const samplerShardSize = traceIdBucketSize * 10;
 
-  const params = {
+  try {
+    const searches = getQueries({
+      query,
+      samplerShardSize,
+      fingerprintBucketSize,
+      isUnfilteredGlobalServiceMap,
+      traceIdBucketSize,
+    });
+
+    const tracesSampleResponse = (
+      await apmEventClient.msearch(
+        'get_trace_sample_ids',
+        ...(isUnfilteredGlobalServiceMap
+          ? [searches.traceSampleIds, searches.traceSampleIdsForSpanLinks]
+          : [searches.traceSampleIds])
+      )
+    ).responses;
+
+    const samples = tracesSampleResponse.flatMap(
+      (response) => response.aggregations?.connections.buckets.map((p) => p.sample) ?? []
+    );
+    // make sure at least one trace per composite/connection bucket is queried
+    const traceIdsWithPriority =
+      samples.flatMap((sample) =>
+        sample.trace_ids.buckets.map((sampleDocBucket, index) => ({
+          traceId: sampleDocBucket.key as string,
+          priority: index,
+        }))
+      ) || [];
```

Comment:
NIT 
`samples.flatMap` should return an empty array in this case we pass an empty array, I don't see it returning anything falsy in any case so do we need the `|| [];`? 
```suggestion
      );
```